hedera-node/src/main/java/com/hedera/services/state/migration/MigrationRecordsManager.java

```diff
@@ -124,6 +143,28 @@ private TransactionBody.Builder synthStakingFundCreate(final long autoRenewPerio
 		return TransactionBody.newBuilder().setCryptoCreateAccount(txnBody);
 	}
 
+	private void publishContractFreeAutoRenewalRecords(final Instant now) {
+		MiscUtils.forEach(accounts.get(), (id, account) -> {
+			if (account.isSmartContract()) {
+				final var contractNum = id.toEntityId();
+				final var newExpiry = account.getExpiry();
+
+				final var receipt = new TxnReceipt();
+				receipt.setAccountId(contractNum);
+
+				final var synthBody = syntheticTxnFactory.synthContractAutoRenew(contractNum.asNum(), newExpiry);
+				final var memo = String.format(CONTRACT_UPGRADE_MEMO, contractNum.num(), newExpiry);
+				final var synthRecord = ExpirableTxnRecord.newBuilder()
+						.setMemo(memo)
+						.setReceipt(receipt)
+						.setConsensusTime(RichInstant.fromJava(now));
```

Comment:
Ah note the `TxnAwareRecordHistorian`  _also_ [ensures unique consensus times](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/records/TxnAwareRecordsHistorian.java#L222) for all child records. So we can remove this as well. (Note we don't ever want to stream two records with the same consensus timetamp.)

Reply:
Removed !